kernel/src/lib/nuumshell/shell.cr

```diff
@@ -0,0 +1,66 @@
+require "./apps/*"
+
+class NuumShell
+  def initialize
+    @user = "root"
+    @pass = "root"
+    @login = false
+    @max_login = 3
+  end
+
+  def run
+    banner
+    while !@login
+      login
+    end
+    loop do
+      print "\n#{@user}@nuumsh#: "
+      command = Keyboard.gets.chomp
+      if command.starts_with?("echo")
```

Comment:
Maybe split the command using `String#split` and only check the first item?
For performance reasons, splitting with a limit of 2 would probably be most efficient.

Maybe something like that:
```crystal
input = Keyboard.gets.chomp
command_parts = input.split 2
if command_parts.size != 0
  command = command_parts[0]
  # ...
end
```

Just an idea though, you don't need to change that if you don't like it.